sheepdog/utils/transforms/__init__.py

```diff
@@ -39,11 +39,38 @@ def parse_list_from_string(value):
         1,2,3 -> [1,2,3]
     """
     items = [x.strip() for x in value.split(",")]
+
+    all_ints = True
     try:
-        items = [float(x) for x in items]
-    except ValueError:
-        pass  # not an array of numbers
-    return items
+        # TODO: Actually pass in and use list_type as the expected type
+        #       and don't try to infer it this way.
+        for item in items:
+            if not float(item).is_integer():
+                all_ints = False
+                break
+    except ValueError as exc:
+        current_app.logger.warning(
+            f"list of values {items} are likely NOT ints or floats so we're leaving "
+            f"them as-is. Exception: {exc}"
+        )
+        return items
+
+    if all_ints:
+        current_app.logger.warning(
+            f"list of values {items} could all be integers, so we are ASSUMING they "
+            "are instead of defaulting to float."
+        )
+        # all can be ints, infer `int` as correct type
+        new_items = [int(float(item)) for item in items]
+    else:
+        current_app.logger.warning(
+            f"list of values {items} are NOT all integers, so we are ASSUMING they "
+            "they are all float by default."
+        )
+        # default to float for backwards compatibility
+        new_items = [float(item) for item in items]
```

Comment:
don't we still need something for not numeric values

Reply:
That gets caught outside this. If the float or int conversion raises an exception (like if it's a string of characters) then we default to just returning the list of strings